contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -2540,6 +2540,143 @@ CREATE OR REPLACE FUNCTION sys.json_query(json_string text, path text default '$
 RETURNS sys.NVARCHAR
 AS 'babelfishpg_tsql', 'tsql_json_query' LANGUAGE C IMMUTABLE PARALLEL SAFE;
 
+-- Function to convert the path format into expected jsonb_set path format using Regular Expression
+CREATE OR REPLACE FUNCTION sys.regex_replace(json_path TEXT)
+RETURNS TEXT
+AS
+$BODY$
+BEGIN
+    SELECT regexp_replace(json_path, '\$\.|]|\$\[' , '' , 'ig') INTO json_path; -- To remove the "$." and "]" sign from the string 
+    SELECT regexp_replace(json_path, '\.|\[' , ',' , 'ig') INTO json_path; -- To replace the "." and "[" with the "," to change into required format
+    SELECT CONCAT('{',json_path,'}') INTO json_path; -- Final required format of path by jsonb_set
+    return json_path;
+END
+$BODY$
+LANGUAGE plpgsql;
+
+--Function to set the append_modifier and create_if_missing flags
+CREATE TYPE falgs_set AS (create_if_missing BOOL, append_modifier BOOL);
+
+CREATE OR REPLACE FUNCTION sys.set_flags(path_json TEXT)
+RETURNS falgs_set
+AS
+$BODY$
+DECLARE
+    result falgs_set;
+    word_count INTEGER;
+    create_if_missing BOOL:='TRUE';
+    mode TEXT;
+    modifier TEXT;
+    append_modifier BOOL:='FALSE';
+BEGIN
+    
+    word_count = LENGTH(path_json) - LENGTH(REPLACE(path_json, ' ', '')) + 1; --To calculate the number of words in the path_json
+    
+    -- This if else block is added to set the create_if_missing and append_modifier flags
+    IF word_count=1 THEN --The path_json has only 1 word
+        create_if_missing='TRUE';
+        append_modifier='FALSE';
+    ELSIF word_count=2 THEN --The path_json has 2 words
+        mode = SPLIT_PART(path_json COLLATE sql_latin1_general_cp1_cs_as, ' ',1);
+        IF mode='append' THEN
+            append_modifier:='TRUE';
+        ELSIF mode='strict' THEN
+            create_if_missing:='FALSE';
+        END IF;
+    ELSIF word_count=3 THEN --The path_json has 3 words
+        modifier = SPLIT_PART(path_json COLLATE sql_latin1_general_cp1_cs_as, ' ',1);
+        mode = SPLIT_PART(path_json COLLATE sql_latin1_general_cp1_cs_as, ' ',2);
+        IF modifier='append' THEN
+            append_modifier:='TRUE';
+        END IF;
+        IF mode='strict' THEN
+            create_if_missing:='FALSE';
+        END IF;
+    END IF;
+    SELECT create_if_missing,append_modifier INTO result.create_if_missing, result.append_modifier;
+    RETURN result;
+END
+$BODY$
+LANGUAGE plpgsql;
+
+
+--JSON_MODIFY
+CREATE OR REPLACE FUNCTION sys.json_modify(in json_string JSONB,in path_json TEXT, in new_value TEXT)
+RETURNS sys.NVARCHAR
+AS
+$BODY$
+DECLARE
+    json_path TEXT;
+    result falgs_set;
```

Comment:
result is not a good variable name for this - makes it seem like we're saving the result in this variable.

Reply:
Have changed the variable name to mode_flags